client/rpc/src/eth.rs

```diff
@@ -827,13 +828,34 @@ where
 		let transaction_hash =
 			H256::from_slice(Keccak256::digest(&rlp::encode(&transaction)).as_slice());
 		let hash = self.client.info().best_hash;
+
+		let id = BlockId::hash(hash);
+		let has_api = match self.client
+			.runtime_api()
+			.has_api::<dyn ConvertTransactionRuntimeApi<B>>(&id) {
+
+			Ok(has_api) => has_api,
+			_ => return Box::pin(future::err(internal_err("cannot access runtime api")))
+			
+		};
+
+		let extrinsic = if has_api {
+			match self.client.runtime_api().convert_transaction(&id, transaction) {
+				Ok(xt) => xt,
+				_ => return Box::pin(future::err(internal_err("cannot access runtime api")))
+			}
+		} else if self.convert_transaction.is_some() {
+			self.convert_transaction.as_ref().unwrap().convert_transaction(transaction)
```

Comment:
On our frontier fork, the client should not use the TransactionConverter, it should rely on the runtime API in any case, because we will update the runtime with this API before any client update.

Reply:
Ok but this should be as close as possible to whatever will go upstream right?

Reply:
I think we can verify it works without it if we the transaction_converter in moonbeam service is `None`.

Reply:
Sorry I'm not awake, you are right